pkg/cmd/cli/nodeagent/server.go

```diff
@@ -333,6 +337,72 @@ func (s *nodeAgentServer) markInProgressCRsFailed() {
 	s.markInProgressPVRsFailed(client)
 }
 
+func (s *nodeAgentServer) markDataUploadsCancel(r *controller.DataUploadReconciler) {
+	// the function is called before starting the controller manager, the embedded client isn't ready to use, so create a new one here
+	client, err := ctrlclient.New(s.mgr.GetConfig(), ctrlclient.Options{Scheme: s.mgr.GetScheme()})
+	if err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to create client")
+		return
+	}
+	if dataUploads, err := r.FindDataUploads(s.ctx, client, s.namespace); err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to find data downloads")
+	} else {
+		for i := range dataUploads {
+			du := dataUploads[i]
+			if du.Status.Phase == velerov2alpha1api.DataUploadPhaseAccepted ||
+				du.Status.Phase == velerov2alpha1api.DataUploadPhasePrepared ||
+				du.Status.Phase == velerov2alpha1api.DataUploadPhaseInProgress {
+				updated := du.DeepCopy()
+				updated.Spec.Cancel = true
+				updated.Status.Message = fmt.Sprintf("found a dataupload with status %q during the node-agent starting, mark it as cancel", du.Status.Phase)
+				du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceled
+				if du.Status.StartTimestamp.IsZero() {
+					du.Status.StartTimestamp = &metav1.Time{Time: r.Clock.Now()}
+				}
+				du.Status.CompletionTimestamp = &metav1.Time{Time: r.Clock.Now()}
+				if err := client.Patch(s.ctx, updated, ctrlclient.MergeFrom(&du)); err != nil {
+					s.logger.WithError(errors.WithStack(err)).Errorf("failed to mark datadownload %q cancel", du.GetName())
+					continue
+				}
+				s.logger.WithField("dataupload", du.GetName()).Warn(du.Status.Message)
+			}
+		}
+	}
+}
+
+func (s *nodeAgentServer) markDataDownloadsCancel(r *controller.DataDownloadReconciler) {
+	// the function is called before starting the controller manager, the embedded client isn't ready to use, so create a new one here
+	client, err := ctrlclient.New(s.mgr.GetConfig(), ctrlclient.Options{Scheme: s.mgr.GetScheme()})
+	if err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to create client")
+		return
+	}
+	if dataDownloads, err := r.FindDataDownloads(s.ctx, client, s.namespace); err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to find data downloads")
+	} else {
+		for i := range dataDownloads {
+			dd := dataDownloads[i]
+			if dd.Status.Phase == velerov2alpha1api.DataDownloadPhaseAccepted ||
```

Comment:
The same as above add a wrapper function to do this cancellation centrally

Reply:
As we discussed, keep the code unchanged until backup deletion really needs one wrapper function